src/set/bytes.rs

```diff
@@ -0,0 +1,130 @@
+//! Serializer codec for serializing a list of byte arrays as a set
+//!
+//! # Usage
+//!
+//! To use, annotate the field with `#[serde(with = "serde_dynamo::set::bytes")]`.
+//!
+//! DynamoDB will return an error if given an empty set. Thus, it may
+//! be beneficial to additionally annotate the field with `#[serde(default)]`
+//! and `#[serde(skip_serializing_if = "<empty check>")]`. This will make sure
+//! that the field is omitted when empty.
+//!
+//! This serializer does not check for duplicate values or an empty set.
+//! If the set contains duplicate values or is empty, DynamoDB will return a
+//! validation error when the attribute value is used.
+//!
+//! # Errors
+//!
+//! The serializer in this module will return an error if:
+//!
+//! * the value does not serialize as a sequence
+//! * the sequence contains any value that is not a byte array
+//!
+//! # Examples
+//!
+//! ```
+//! use serde_bytes::ByteBuf;
+//! use serde_derive::{Serialize, Deserialize};
+//! use serde_dynamo::{Item, AttributeValue};
+//!
+//! #[derive(Serialize, Deserialize)]
+//! struct MyStruct {
+//!     #[serde(with = "serde_dynamo::set::bytes")]
+//!     #[serde(default, skip_serializing_if = "Vec::is_empty")]
+//!     data: Vec<ByteBuf>,
+//! }
+//!
+//! let my_struct = MyStruct {
+//!     data: vec![
+//!         ByteBuf::from(b"hello".to_vec()),
+//!         ByteBuf::from(b"world".to_vec())
+//!     ].into(),
+//! };
+//!
+//! let serialized: Item = serde_dynamo::to_item(&my_struct).unwrap();
+//! assert_eq!(
+//!     serialized["data"],
+//!     AttributeValue::Bs(vec![b"hello".to_vec(), b"world".to_vec()])
+//! );
+//! ```
+
+pub(super) static NEWTYPE_SYMBOL: &str = "\u{037E}BYTESSET\u{037E}";
+
+#[inline]
+pub(crate) fn should_serialize_as_bytes_set(name: &str) -> bool {
+    std::ptr::eq(name, NEWTYPE_SYMBOL)
+}
```

Comment:
Have you seen this approach before in other production crates? I'm not familiar enough with this lower level stuff to know if this is definitely going to work into the future. I'd feel a lot more comfortable if I know other crates were doing the same thing.

Reply:
This is certainly not a _common_ usage, but it is called out as a potential solution in this [serde issue comment](https://github.com/serde-rs/serde/issues/529#issuecomment-243486957), as a workaround for a lack of stable specialization in Rust. I do this static pointer equality as a further step to prevent an unintentional name collision.

Reply:
Awesome. I think that's compelling enough to consider this a viable option.